yarn-project/aztec-rpc/src/simulator_oracle/index.ts

```diff
@@ -21,18 +21,21 @@ export class SimulatorOracle implements DBOracle {
   ) {}
 
   /**
-   * Retrieve the secret key associated with a specific address.
+   * Retrieve the secret key associated with a specific public key.
    * The function only allows access to the secret keys of the transaction creator,
    * and throws an error if the address does not match the public key address of the key pair.
    *
    * @param _contractAddress - The contract address. Ignored here. But we might want to return different keys for different contracts.
-   * @param txOriginAddress - The address of an account.
+   * @param pubKey - The public key of an account.
    * @returns A Promise that resolves to the secret key as a Buffer.
    * @throws An Error if the input address does not match the public key address of the key pair.
    */
-  getSecretKey(_contractAddress: AztecAddress, txOriginAddress: AztecAddress): Promise<Buffer> {
-    if (!txOriginAddress.equals(this.address)) {
-      throw new Error('Only allow access to the secret keys of the tx creator.');
+  getSecretKey(_contractAddress: AztecAddress, pubKey: Point): Promise<Buffer> {
+    const thisPubKey = this.keyPair.getPublicKey();
+    if (!thisPubKey.equals(pubKey)) {
+      throw new Error(
+        `Only allow access to the secret keys of the tx creator (requested keys for ${pubKey}, expected ${thisPubKey}).`,
+      );
     }
     return this.keyPair.getPrivateKey();
   }
```

Comment:
@sirasistant this method now requires a pubkey rather than an address, which was already being passed in from Noir. Can you check it's ok to change it?

Reply:
Yeah, it was only using address because previously we indexed accounts by address. The simulator / oracle changes LGTM :rocket: